### ISSUE TITLE
when building project in Karavan - Timeout waiting to connect to the Maven daemon

### ISSUE THREAD

**Issue by git-josip** [apache/camel-karavan#827]
When I try to build spring boot camel project in Karavan I get error `Timeout waiting to connect to the Maven daemon.` 
I need to rerun deploy step may times to aucceed. 

 This is complete output from tekton pipeline run:
```
Cloning into '/scripts'...
jbang -Dcamel.jbang.version=3.20.3 camel@apache/camel export --local-kamelet-dir=/scripts/kamelets
Generating fresh run data
Exporting as Spring Boot project to: .
Exception in thread "main" org.mvndaemon.mvnd.common.DaemonException$ConnectException: Timeout waiting to connect to the Maven daemon.
Daemon id: d121ebb0
  log file: /root/.m2/mvnd/registry/0.9.0/daemon-d121ebb0.log
----- Last  200 lines from daemon log file - /root/.m2/mvnd/registry/0.9.0/daemon-d121ebb0.log -----

----- End of the daemon log file -----
  output: /root/.m2/mvnd/registry/0.9.0/daemon-d121ebb0.out.log
----- Last  200 lines from daemon output - /root/.m2/mvnd/registry/0.9.0/daemon-d121ebb0.out.log -----
Starting daemon process: id = d121ebb0, workingDir = /scripts/hzv, daemonArgs: /opt/java/openjdk/bin/java -classpath /opt/mvnd/lib/mvnd-common-0.9.0.jar:/opt/mvnd/lib/mvnd-agent-0.9.0.jar -javaagent:/opt/mvnd/lib/mvnd-agent-0.9.0.jar -Dmvnd.home=/opt/mvnd -Dmaven.home=/opt/mvnd -Dmaven.conf=/opt/mvnd/conf -Dmvnd.java.home=/opt/java/openjdk -Dlogback.configurationFile=/opt/mvnd/conf/logging/logback-mvnd.xml -Dmvnd.id=d121ebb0 -Dmvnd.daemonStorage=/root/.m2/mvnd/registry/0.9.0 -Dmvnd.registry=/root/.m2/mvnd/registry/0.9.0/registry.bin -Dmvnd.socketFamily=inet -Dmvnd.home=/opt/mvnd -Djdk.java.options=--add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/sun.net.www.protocol.jar=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Dmvnd.noDaemon=false -Dmvnd.debug=false -Dmvnd.debug.address=8000 -Dmvnd.idleTimeout=3h -Dmvnd.keepAlive=100ms -Dmvnd.extClasspath= -Dmvnd.coreExtensions= -Dmvnd.enableAssertions=false -Dmvnd.expirationCheckDelay=10s -Dmvnd.duplicateDaemonGracePeriod=10s -Dmvnd.socketFamily=inet org.mvndaemon.mvnd.common.MavenDaemonNOTE: Picked up JDK_JAVA_OPTIONS: --add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/sun.net.www.protocol.jar=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED
----- End of the daemon output -----

	at org.mvndaemon.mvnd.client.DaemonConnector.startDaemon([DaemonConnector.java:331](http://daemonconnector.java:331/))
	at org.mvndaemon.mvnd.client.DaemonConnector.connect([DaemonConnector.java:124](http://daemonconnector.java:124/))
	at org.mvndaemon.mvnd.client.DefaultClient.execute([DefaultClient.java:327](http://defaultclient.java:327/))
	at org.mvndaemon.mvnd.client.DefaultClient.main([DefaultClient.java:153](http://defaultclient.java:153/))
```

**Comment by mgubaidullin** [comment]
Looks like you are using version 3.20.x
Could you, please, try the latest 3.21.0

**Comment by git-josip** [comment]
@mgubaidullin thanks. It is working in `3.21.0` version. 

I had to do some manual changes in Tekton pipeline task which I am using as I am not sure how to update those. 

I have done following: 
- edit `karavan` pod and update image to `3.21.0` version. 
- - `kubectl edit pods karavan-8597fb6897-k9942  -n karavan`
- - `image: ghcr.io/apache/camel-karavan:3.21.0` 

Also I have update Tekton task `karavan-task-dev-spring-boot` as I am using camel spring type. 
-- ` kubectl edit  tasks karavan-task-dev-spring-boot -n karavan` 
-- `image: ghcr.io/apache/camel-karavan-builder:3.21.0` 


I have noticed `mvnd` is not used anymore, `mvn` is used so I have also changed the command in task from `/opt/maven/bin/mvn` to just plain `mvn` . 

Maybe there is better way to update all of these.
